### PR TITLE
[release-1.16] Update workspace to support Envoy CVE fixes (#72)

### PR DESCRIPTION
**Pull request by GregHanson**
* Update workspace to support Envoy CVE fixes

* Fix build compilation

* Fix testdata

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by dhawton — APPROVED** [review]

**Comment by GregHanson** [comment]
leaving` do-not-merge` label pending follow-up discussion

File: Makefile.core.mk [hunk 1]
```diff
@@ -155,7 +155,6 @@ check:
 lint: lint-copyright-banner format-go lint-go tidy-go lint-scripts
 	@scripts/check-repository.sh
 	@scripts/check-style.sh
-	@scripts/verify-last-flag-matches-upstream.sh
```
Comment (jacob-delgado) [inline Makefile.core.mk:158]: actually we do want this for the non-private branches

File: WORKSPACE [hunk 2]
```diff
@@ -38,12 +38,12 @@ bind(
...
 # Note: this is needed by release builder to resolve envoy dep sha to tag.
-# Commit date: 2023-03-24
-ENVOY_SHA = "b39e09a0fb0435aea66c4240c3cd7bb4afa377c0"
+# Commit date: 2023-03-28
+ENVOY_SHA = "b07cce42ccaf39a37d2f1ac54995cc048d4db92d"
 
-ENVOY_SHA256 = "bc3430bfd9d90229eaf2ab9d3fdc13c49ff5d455b0b56897bff68086eab7006c"
+ENVOY_SHA256 = "3a449d355dc0cedfdacb83553bebf63f45b1e24f478fd8aa4de67d4ffa4c7de7"
 
-ENVOY_ORG = "envoyproxy"
```
Comment (jacob-delgado) [inline WORKSPACE:46]: don't modify this file either

File: testdata/stackdriver/utf8_client_access_log_entry.yaml.tmpl [hunk 3]
```diff
@@ -25,6 +25,5 @@ labels:
   upstream_cluster: "server-outbound-cluster"
   route_name: client_route
   response_details: "via_upstream"
-  x-envoy-original-path: va lue
```
Comment (jacob-delgado) [inline testdata/stackdriver/utf8_client_access_log_entry.yaml.tmpl:28]: these are the only 2 files you need to change

**Pull request closed without merging**